query-engine/core/src/query_ast/write.rs

```diff
@@ -151,19 +160,41 @@ pub struct UpdateRecord {
     pub args: WriteArgs,
 }
 
+impl UpdateRecord {
```

Comment:
So, all mutations shouldn't implement `set_selectors` because not all mutations _have_ use a `RecordFilter`. I thought about making this an interface and implement it on the mutations that needed it but I didn't see much added value besides adding indirection. Still, I converted it to a trait and implemented it on the relevant mutations. We'll see how that evolves 👍